Runtime/Core/ShrinkablePolygon.cs

```diff
@@ -265,15 +275,15 @@ public void ComputeAspectBasedShrinkDirections(in AspectData aspectData)
                 m_Points[i] = p;
             }
 
-            // update m_State, if change happened based on the cached shrink directions
-            for (var index = 0; index < s_shrinkDirectionsCache.Count; index++)
-            {
-                if (s_shrinkDirectionsCache[index] != m_Points[index].m_ShrinkDirection)
-                {
-                    m_State++; // m_State change when even one shrink direction has been changed
-                    break;
-                }
-            }
+            // // update m_State, if change happened based on the cached shrink directions
+            // for (var index = 0; index < s_shrinkDirectionsCache.Count; index++)
+            // {
+            //     if (s_shrinkDirectionsCache[index] != m_Points[index].m_ShrinkDirection)
+            //     {
+            //         m_State++; // m_State change when even one shrink direction has been changed
+            //         break;
+            //     }
+            // }
```

Comment:
We don't care anymore if a direction changes?  Or does that only happen if there is an intersection?

Reply:
We don't because direction change happens after intersection happened.